sormas-e2e-tests/src/test/java/org/sormas/e2etests/helpers/WebDriverHelpers.java

```diff
@@ -373,7 +374,43 @@ public void waitUntilNumberOfElementsIsReduceToGiven(By selector, int given) {
 
   public String getCheckedOptionFromHorizontalOptionGroup(By options) {
     waitUntilIdentifiedElementIsPresent(options);
-    return baseSteps.getDriver().findElement(options).findElement(SELECTED_RADIO_BUTTON).getText();
+    if (baseSteps.getDriver().findElement(options).findElements(SELECTED_RADIO_BUTTON).size()
+        == 0) {
+      return null;
+    } else {
+      return baseSteps
+          .getDriver()
+          .findElement(options)
+          .findElement(SELECTED_RADIO_BUTTON)
+          .getText();
+    }
+  }
+
+  public boolean isSingleCssInputChecked(By cssInput) {
+    waitUntilIdentifiedElementIsPresent(cssInput);
+    scrollToElement(cssInput);
+    try {
+      boolean value =
+          (baseSteps.getDriver().findElement(cssInput).getAttribute("checked").equals("true"))
+              ? true
+              : false;
+      return value;
+    } catch (Exception e) {
+      return false;
+    }
+  }
+
+  public String getValueForSingleCssInputIfChecked(By cssInput) {
+    if (isSingleCssInputChecked(cssInput)) {
+      String checkedElementCssSelector =
+          StringUtils.getMatchedGroupByIndexFromAString(
+              baseSteps.getDriver().findElement(cssInput).toString(),
+              "(.*css selector:\\s)(.*)(\\s>\\sinput.*)",
+              2);
+      return baseSteps.getDriver().findElement(By.cssSelector(checkedElementCssSelector)).getText();
+    } else {
+      return null;
+    }
   }
```

Comment:
Please use KISS, it's very simple an input has or not the `checked` property.

Reply:
the problem is that if using an assert only for attribute: checked to be true or null, there could be some mistakes in locators.

using that condition for getting the text from the checked input assures that you checks the right element.

also it does not break the pattern we are using at the moment. 

@Tamas, please let me know if you are sure that it should be only checked true/null?

Reply:
We will use this one only for the elements with the `checked` properties

Reply:
my logic here is:

    1. get the status checked of the input using isSingleCssInputChecked and parse it in true or false;
    2. use the getValueForSingleCssInputIfChecked to parse that boolean result from isSingleCssInputChecked and return the text near the selected element when true or null otherwise .

I'm doing it by managing the patterns. this will be work as long as the develop is using the same pattern for designing the inputs (they are inherit this pattern in sormas arhitechture) and we are using the ccsSelectors over all the project. this can be extended to other selectors, too.